### PR TITLE
87 notifications

### PR DESCRIPTION
**Pull request by johlton**
First version. Test messages from within /core/actions/SingleFileConvertAction.java

### CODE REVIEW

**Comment by johlton** [comment]
To-Do: make sure that the notification buffer can be passed to other parts of the application which could get cumbersome pretty fast..

File: kitodo-mediaserver-core/src/main/java/org/kitodo/mediaserver/core/util/Notifier.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+    /**
+     * Appends a message to a notification, including timestamp.
+     *
+     * @param buffer The notification buffer
+     * @param message the message
+     * @param includeTimestamp Whether to include a timestamp with each message
+     */
+    @Override
+    public void add(StringBuffer buffer, String message, boolean includeTimestamp) {
```
Comment (pbroman) [inline kitodo-mediaserver-core/src/main/java/org/kitodo/mediaserver/core/util/Notifier.java]: You are not using the boolean param here. With this solution, the method should rather be called addWithTimestamp and have no includeTimestamp param.

File: kitodo-mediaserver-core/src/main/java/org/kitodo/mediaserver/core/config/NotifierProperties.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+import org.springframework.context.annotation.Configuration;
+
+/**
+ * Properties class for the conversion configuration.
+ */
+@Configuration
+@ConfigurationProperties(prefix = "notifier")
+public class NotifierProperties {
+
+    private String level;
```
Comment (pbroman) [inline kitodo-mediaserver-core/src/main/java/org/kitodo/mediaserver/core/config/NotifierProperties.java]: As far as I can see, the notifier level is never used.

File: kitodo-mediaserver-core/src/main/java/org/kitodo/mediaserver/core/util/Notifier.java [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+        mailSender.send(message);
+    }
+
+    /**
+     * Returns string with current datetime in a configured format.
+     */
+    protected String getTime() {
+        String format = notifierProperties.getTimestampFormat();
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern(format)
+                .withLocale(Locale.GERMANY)
```
Comment (ponchofiesta) [inline kitodo-mediaserver-core/src/main/java/org/kitodo/mediaserver/core/util/Notifier.java]: Locale and maybe time zone should not be hardcoded.

File: kitodo-mediaserver-core/src/main/resources/config/default.yml [hunk 4]
```diff
@@ -15,6 +15,15 @@ spring:
     password: kitodo
   main:
     banner-mode: "off"
+  mail:
+    host: localhost
+    username:
+    password:
+    smtp:
```
Comment (ponchofiesta) [inline kitodo-mediaserver-core/src/main/resources/config/default.yml]: Is this key used anywhere? (it should)
Also there should be a possibility to use TLS implicitly and to set a port number.

Comment (johlton) [reply kitodo-mediaserver-core/src/main/resources/config/default.yml]: It's used automatically by Spring for JavaMail. You could use port 465 to get implicit TLS. And therefore, yes, you are right, port setting is missing.

File: kitodo-mediaserver-core/src/main/java/org/kitodo/mediaserver/core/config/NotifierProperties.java (hunk 2 above)
Comment (johlton) [reply kitodo-mediaserver-core/src/main/java/org/kitodo/mediaserver/core/config/NotifierProperties.java]: Will remove it with the next commit.

**Pull request closed without merging**